monai/apps/deepgrow/dataset.py

```diff
@@ -0,0 +1,270 @@
+from monai.utils import GridSampleMode
+
+
+def create_dataset(
+    datalist,
+    output_dir,
+    dimension,
+    pixdim,
+    keys=("image", "label"),
+    base_dir=None,
+    limit=0,
+    relative_path=False,
+    transforms=None,
+) -> List[Dict]:
+    """
+    Utility to pre-process and create dataset list for Deepgrow training over on existing one.
+    The input data list is normally a list of images and labels (3D volume) that needs pre-processing
+    for Deepgrow training pipeline.
+
+    Args:
+        datalist: A generic dataset with a length property which normally contains a list of data dictionary.
+            For example, typical input data can be a list of dictionaries::
+
+                [{'image': 'img1.nii', 'label': 'label1.nii'}]
+
+        output_dir: target directory to store the training data for Deepgrow Training
+        pixdim: output voxel spacing.
+        dimension: dimension for Deepgrow training.  It can be 2 or 3.
+        keys: Image and Label keys in input datalist.  Defaults to 'image' and 'label'
+        base_dir: base directory in case related path is used for the keys in datalist.  Defaults to None.
+        limit: limit number of inputs for pre-processing.  Defaults to 0 (no limit).
+        relative_path: output keys values should be based on relative path.  Defaults to False.
+        transforms: explicit transforms to execute operations on input data.
+
+    Raises:
+        ValueError: When ``dimension`` is not one of [2, 3]
+        ValueError: When ``datalist`` is Empty
+
+    Example::
+
+        datalist = create_dataset(
+            datalist=[{'image': 'img1.nii', 'label': 'label1.nii'}],
+            base_dir=None,
+            output_dir=output_2d,
+            dimension=2,
+            keys=('image', 'label')
+            pixdim=(1.0, 1.0),
+            limit=0,
+            relative_path=True
+        )
+
+        print(datalist[0]["image"], datalist[0]["label"])
+    """
+
+    if dimension not in [2, 3]:
+        raise ValueError("Dimension can be only 2 or 3 as Deepgrow supports only 2D/3D Training")
+
+    if not len(datalist):
+        raise ValueError("Input Datalist is empty")
+
+    if not isinstance(keys, list) and not isinstance(keys, tuple):
+        keys = [keys]
+
+    transforms = _default_transforms(keys, pixdim) if transforms is None else transforms
+    new_datalist = []
+    for idx in range(len(datalist)):
+        if limit and idx >= limit:
+            break
+
+        image = datalist[idx][keys[0]]
+        label = datalist[idx].get(keys[1]) if len(keys) > 1 else None
+        if base_dir:
+            image = os.path.join(base_dir, image)
+            label = os.path.join(base_dir, label) if label else None
+
+        image = os.path.abspath(image)
+        label = os.path.abspath(label) if label else None
+
+        logging.info("Image: {}; Label: {}".format(image, label if label else None))
+        if dimension == 2:
```

Comment:
we can prepare data for deepgrow training from both 2D and 3D from the given 3D nifti volume.  In case of 2D, all the slices gets flattened against all the regions in the label.  So, we need user choice for which dimension, he/she likes to prepare the data for deepgrow training

Reply:
OK, understood.